nexus_keycode/test/protocols/test_small.py

```diff
@@ -214,6 +215,215 @@ def test_compressed_message_bits__unlock__output_correct(self):
+
+        # All '0' bits
+        bits = bitstring.Bits(bin='0b00000000000000000000000000')
+        message = protocol.PassthroughSmallMessage(bits)
+        self.assertEqual(message.compressed_message_bits[0:28].bin, "0000000100000000000000000000")
+        self.assertEqual("124 325 434 222 222", message.to_keycode())
+
+        # alternating
+        bits = bitstring.Bits(bin='0b01010101010101010101010101')
+        message = protocol.PassthroughSmallMessage(bits)
+        self.assertEqual(message.compressed_message_bits[0:28].bin, "0101010101010101010101010101")
+        self.assertEqual("132 423 253 333 333", message.to_keycode())
+
+        # 10-long pattern
+        bits = bitstring.Bits(bin='0b11100110101110011010111001')
+        message = protocol.PassthroughSmallMessage(bits)
+        self.assertEqual(message.compressed_message_bits[0:28].bin, "1110010110101110011010111001")
+        self.assertEqual("123 534 332 344 543", message.to_keycode())
+
+    def test_compressed_message_bits__invalid_length__raises(self):
+        # All '1' bits (27 bits)
+        bits = bitstring.Bits(bin='0b111111111111111111111111111')
+        with self.assertRaises(ValueError):
+            protocol.PassthroughSmallMessage(bits)
+
+        # All '1' bits (25 bits)
+        bits = bitstring.Bits(bin='0b1111111111111111111111111')
+        with self.assertRaises(ValueError):
+            protocol.PassthroughSmallMessage(bits)
+
+
+class TestExtendedSmallMessage(TestCase):
+    def test_init__invalid_command_type__raises(self):
+        with self.assertRaises(ValueError):
+            protocol.ExtendedSmallMessage(
+                220,
+                id_=102,
+                days=30,
+                secret_key=b"\xab" * 16
+            )
+
+    def test_init__valid_message_types__expected_value_returned(self):
+        secret_key = b"\xab" * 16
+        message = protocol.ExtendedSmallMessage(
+            protocol.ExtendedSmallMessageType.SET_CREDIT_WIPE_RESTRICTED_FLAG,
+            id_=102,
+            days=30,
+            secret_key=secret_key)
+
+        body_bits = message.body
+        # first bit = app ID
+        self.assertEqual(1, body_bits[0:1].uint)
+        # type code
+        self.assertEqual(
+            protocol.ExtendedSmallMessageType.SET_CREDIT_WIPE_RESTRICTED_FLAG.value[0],
+            body_bits[1:4].uint
+        )
+        # LSB 2 bits of message ID (0b10 for 102)
+        self.assertEqual(2, body_bits[4:6].uint)
+        # increment ID for days = 30
+        self.assertEqual(29, body_bits[6:14].uint)
+
+        computed_auth = protocol.ExtendedSmallMessage._compute_auth(
+            102,
+            protocol.ExtendedSmallMessageType.SET_CREDIT_WIPE_RESTRICTED_FLAG,
+            body_bits[4:14],  # body of inner 'extended' message
+            secret_key)
+
+        self.assertEqual(computed_auth, body_bits[14:26].uint)
+
+        # fixed MAC (matches 223 345 below)
```

Comment:
This changed? Comment no longer makes sense.

Reply:
addressing